README.md

```diff
@@ -1,9 +1,148 @@
+
+You don't need to run the web server locally, you can just create a [kubernetes deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) using the `web-server_deployment.yml` manifest file and the web server will be deployed to your cluster:
+
+```
+oc apply -f web-server_deployment.yml
+```
+
+Optionally, you can [run the web server locally](#optional-running-the-web-server-locally), see section below.
+
+### Create a Service
+
+To expose the application you have to create a [kubernetes service](https://kubernetes.io/docs/concepts/services-networking/service/):
+
+```
+oc apply -f web-server_service.yml
+```
+
+### Create an OpenShift Route
+Finally, to create a public URL to access the application you should create an OpenShift route:
+
+```
+oc apply -f web-server_route.yaml
+```
+
+**Congratulations!**, now you can visit the application at: http://k8sinitiative.apps.dev-eng-ocp4-5.dev.3sca.net/
+
+**Important notes:**
+
+- The [host field](https://github.com/3scale/k8sapp-initiative/blob/master/web-server_route.yaml#L7) in `web-server_route.yaml` is currently pointing to a 3scale owned domain. You should change this to fit your needs, maybe using an environment variable accessible in your cluster.
+
+- *Routes* are specific to OpenShift, for alternatives see [alternatives to OpenShift](#optional-alternatives-to-openshift-routes) below.
+
+### [OPTIONAL] Alternatives to OpenShift Routes
+
+If you don't have access to OpenShift or you prefer to use a kubernetes-only solution, you shoud create your own Ingress object.
+
+Here are some useful documentation you can check:
+
+[Kubernetes Ingress documentation](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+
+[Kubernetes Ingress vs OpenShift Route](https://www.openshift.com/blog/kubernetes-ingress-vs-openshift-route)
+
+### [OPTIONAL] Running the web server locally
+
+#### Configuration
+
+The web server is designed to run in the context of a kubernetes cluster and needs access to some special configuration and environment variables such as KUBERNETES_SERVICE_HOST, KUBERNETES_SERVICE_PORT and PORT. 
+
+If you want to run the web server locally **is your responsiblity to setup this configuration**.
+
+We have created a [controller](https://github.com/3scale/k8sapp-initiative/tree/master/controller) using the offical **Go client** for talking to the kubernetes cluster, you can find more info at https://github.com/kubernetes/client-go and https://pkg.go.dev/k8s.io/client-go.
+
+**Build and tag container image:**
+
+Change directory:
+```
+cd web-server/
+```
+Build and tag the image:
+```
+DOCKER_BUILDKIT=1 docker build -f Dockerfile -t <YOUR_USER>/<IMAGE_NAME> .
+```
+**Run Go web server:**
+
+Change directory:
+```
+cd pkg/
+```
+Run server:
+
+```
+go run main.go
```

Comment:
This specific step won't work, this would need the `PORT` env, but most importantly, it's coded to run within a cluster, not in your local